common/src/main/java/nl/tudelft/trustchain/common/util/VotingHelper.kt

```
@@ -156,24 +136,124 @@ class VotingHelper(
+        val yescount = count.first
+        val nocount = count.second
+
+        val mode = try {
+            VotingMode.valueOf(voteMode)
+        } catch (e: IllegalArgumentException) {
+            handleInvalidVote(block, "The VotingMode was invalid, its value was $voteMode.")
+        }
+
+        return if (mode == VotingMode.THRESHOLD) {
+            if (threshold == -1) {
+                throw IllegalArgumentException("VotingMode was set to THRESHOLD but threshold was not specified as function argument.")
+            }
+
+            yescount >= threshold
+        } else {
+            yescount + nocount == voters.size
+        }
+    }
+
+    /**
+     * Return the string value for a JSON tag in a voting block.
+     */
+    private fun getVoteBlockAttributesByKey(block: TrustChainBlock, key: String): String {
+
+        // Skip all blocks which are not voting blocks
+        // and don't have a 'message' field in their transaction.
+        if (block.type != votingBlock || !block.transaction.containsKey("message")) {
+            handleInvalidVote(block, "Block was not a voting block or did not contain a 'message' field in its transaction.")
+        }
+
+        val voteJSON = try {
+            JSONObject(block.transaction["message"].toString())
+        } catch (e: JSONException) {
+            handleInvalidVote(block, "Voting block did not contain proper JSON.")
+        }
+
+        try {
+            return voteJSON.get(key).toString()
+        } catch (e: JSONException) {
+            handleInvalidVote(block, "Voting JSON did not have a value for key '$key'.")
+        }
+    }
+
+    /**
+     * Throw an exception when an invalid vote is encountered.
+     */
+    private fun handleInvalidVote(block: TrustChainBlock, errorType: String): Nothing {
+        Log.e("vote_debug", "Encountered an invalid voting block with ID ${block.blockId}." +
+            "The reason for invalidity was: $errorType")
+        throw Exception(errorType)
```

Do you really want to throw an exception instead of fail gracefully?

---

Yes. It is currently beyond the scope of the PR. An invalid vote should be handled as a separate issue.

---

Okay for now, but I would like to see this fixed in the near future

---

I do not agree with this actually, by not doing this, this pr is essentially breaking current functionality. If I click on an old vote proposal, all crashes withouth telling me why (through the UI).

---

I've surrounded it in a try catch for now, but would like to see it improved in the future.

---

Alright, we should definitely fix this in a new pr, for now okay..